utils/elastic.ts

```diff
@@ -27,46 +27,55 @@ const BULKSIZE = 5000;
 type ElasticIndex = {
   name: string;
   mapping: any;
-  alias: string;
 };
 
 export class Elastic {
   public CLASS_ALIAS: string;
   public EMPLOYEE_ALIAS: string;
 
-  private classIndex: ElasticIndex;
-  private employeeIndex: ElasticIndex;
-  private indexes: ElasticIndex[];
-  private initializing: Promise<void>;
+  private indexes: Record<string, ElasticIndex>;
 
   constructor() {
     // Because we export an instance of this class, put the constants on the instance.
     this.CLASS_ALIAS = "classes";
     this.EMPLOYEE_ALIAS = "employees";
 
-    this.classIndex = {
+    const classIndex = {
       name: "",
       mapping: classMap,
-      alias: this.CLASS_ALIAS,
     };
-    this.employeeIndex = {
+    const employeeIndex = {
       name: "",
       mapping: employeeMap,
-      alias: this.EMPLOYEE_ALIAS,
     };
-    this.indexes = [this.classIndex, this.employeeIndex];
+    this.indexes = {};
+    this.indexes[this.CLASS_ALIAS] = classIndex;
+    this.indexes[this.EMPLOYEE_ALIAS] = employeeIndex;
   }
 
-  // This method fetches the exact index name since they are now dynamically named green or blue
-  async fetchIndexNames() {
-    const isClassesBlue = await this.doesIndexExist("classes_blue");
-    const employeesBlue = await this.doesIndexExist("employees_blue");
-
-    const classesName = isClassesBlue ? "classes_blue" : "classes_green";
-    const employeesName = employeesBlue ? "employees_blue" : "employees_green";
+  async fetchIndexNames(): Promise<void> {
+    const aliases = Object.keys(this.indexes);
+    await Promise.all(
+      aliases.map(async (alias) => await this.fetchIndexName(alias))
+    );
+  }
 
-    this.classIndex.name = classesName;
-    this.employeeIndex.name = employeesName;
+  // This method fetches the exact index name since they are now dynamically named green or blue
+  async fetchIndexName(aliasName: string): Promise<void> {
+    const { mapping } = this.indexes[aliasName];
+    const isBlue = await this.doesIndexExist(`${aliasName}_blue`);
+    const isGreen = await this.doesIndexExist(`${aliasName}_green`);
+
+    // if neither index exists, create a new index
+    if (!isBlue && !isGreen) {
+      const indexName = `${aliasName}_blue`;
+      await this.createIndex(indexName, mapping);
+      await this.createAlias(indexName, aliasName);
+    } else if (isBlue) {
+      this.indexes[aliasName].name = `${aliasName}_blue`;
+    } else {
+      this.indexes[aliasName].name = `${aliasName}_green`;
+    }
```

Comment:
[nit] I'm not a fan of way that in `doesIndexExist(NAME)` and `this.indexes[aliasName].name = NAME`, `NAME` is not a single variable. This could lead to typos down the line (but to be fair, I doubt we'll touch this often, which is why this is just a nit)

```suggestion
    const indexNames = [`${aliasName}_blue`, `${aliasName}_green`];
    
    for (const indexName of indexNames) {
      if (await this.doesIndexExist(indexName)) {
        this.indexes[aliasName].name = indexName;
        return;
      }
    }
    
    // if neither index exists, create a new index
    const indexName = indexNames[0];
    await this.createIndex(indexName, mapping);
    await this.createAlias(indexName, aliasName);
```

Reply:
I like this, I also had the same thought but couldn't think of a reasonable solution at the time